File: WebApp_Restautant/ClientApp/src/app/factura/factura.ts
export interface IDetalle {
  plato: string;
  valor: string;
}

export interface IFactura {
  idSupervisor: string;
  idCliente: string;
  nombres: string;
  apellidos: string;
  direccion: string;
  telefono: string;
  nroMesa: number;
  idMesero: number;
  detalles: IDetalle[];
}

File: WebApp_Restautant/Models/Factura.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WebApp_Restautant.Models;

namespace WebApp_Restautant
{
    public class Factura
    {
        public string IdSupervisor { get; set; }
        public string IdCliente { get; set; }
        public string Nombres { get; set; }
        public string Apellidos { get; set; }
        public string Direccion { get; set; }
        public string Telefono { get; set; }
        public int NroMesa { get; set; }
        public int IdMesero { get; set; }
        public IEnumerable<Detalle> Detalles { get; set; }

        public Factura() {}
    }
}